### PR TITLE
Improve documentation

### LINKED ISSUE #14: Add documentation to explain how to generate an SDK using swagger codegen
This should start as a markdown file in the `service` repository, that gets linked to from the front-end interfaces.

### PR DESCRIPTION
**Pull request by slifty**
This PR adds more detail to the instructions on how to use the code generators.

Resolves #14

### CODE REVIEW

**Comment by slifty** [comment]
Should add in generic instructions here (at least one sentence / a link to their docs on how to use codegen without using our custom templates)

File: README.md [hunk 1]
```diff
@@ -19,13 +19,32 @@ To generate the SDKs supported by this project, you will need:
...
 ```shell
-> cp YOUR_SWAGGER_SPEC_LOCATION ./openapi.json
 > ./gradlew generateSwaggerCode
 ```
 
 The generated code will now exist in the `build` directory.
-Each language target will have its own directory inside of `build` (e.g. `build/typescript`).
-Some languages might have additional build steps defined in their build's README.md file.
+
+Each language target will have its own directory inside of `build`, and you can follow the readme instructions in that directory with additional steps related to that code base.
```
Comment (reefdog) [inline README.md]: Should add in generic instructions here (at least one sentence / a link to their docs on how to use codegen without using our custom templates)

Comment (slifty) [reply README.md]: oh this is a great point

(Also whoops -- I had local changes I apparently hadn't pushed before requesting review!  Please check your inbox for an email asking you to re-review).

Comment (reefdog) [reply README.md]: I _only_ make great points.

File: README.md [hunk 2]
```diff
@@ -2,30 +2,48 @@
...
+
+- Typescript: `build/typescript/README.md`.
```
Comment (reefdog) [inline README.md]: Dan you're going to laugh but I genuinely think this line would be better without the period at the end

File: README.md [hunk 3]
```diff
@@ -2,30 +2,48 @@
 
 This repository houses the infrastructure for generating PDC SDKs.
 
-We're using a project called [swagger-codegen](https://github.com/swagger-api/swagger-codegen) to power our SDK, and this repository provides tooling and templates
-to make sdk generation easier in particular target languages. The codegen project supports a [many languages](https://github.com/swagger-api/swagger-codegen/tree/master/modules/swagger-codegen/src/main/resources)
-beyond the ones explicitly defined here.
+We're using a project called [swagger-codegen](https://github.com/swagger-api/swagger-codegen) to power our SDK, and this repository provides tooling and templates to make sdk generation easier in specifically supported languages.
```
Comment (reefdog) [inline README.md]:
```suggestion
We're using a project called [swagger-codegen](https://github.com/swagger-api/swagger-codegen) to power our SDK, and this repository provides tooling and templates to make SDK generation easier in specifically supported languages.
```

File: README.md (hunk 2 above)
Comment (reefdog) [inline README.md]:
```suggestion
- TypeScript: `build/typescript/README.md`
```

File: README.md [hunk 4]
```diff
@@ -2,30 +2,48 @@
...
 ## Prequisites
 
 - JRE (18+)
 
-## Generating sdks
-To generate the SDKs supported by this project, you will need:
+## Generating SDKs
 
-1. A [swagger](https://swagger.io/) specification located at the project root and named `openapi.json`. The current swagger specification for the PDC can be found in the [service repository](https://github.com/PhilanthropyDataCommons/service/blob/main/src/openapi.json).
+To generate the SDKs supported by this project, you will need a [swagger](https://swagger.io/) specification located at the project root and named `openapi.json`. The current swagger specification for the PDC can be found in the [service repository](https://github.com/PhilanthropyDataCommons/service/blob/main/src/openapi.json).
```
Comment (reefdog) [inline README.md]:
```suggestion
To generate the SDKs supported by this project, you will need a [Swagger](https://swagger.io/) specification located at the project root and named `openapi.json`. The current Swagger specification for the PDC can be found in the [service repository](https://github.com/PhilanthropyDataCommons/service/blob/main/src/openapi.json).
```

File: README.md [hunk 5]
```diff
@@ -2,30 +2,48 @@
...
+git clone git@github.com:PhilanthropyDataCommons/sdk.git
+```
+
+2. Change directory:
+
+```shell
+cd sdk
+```
+
+3. Copy your swagger specification
```
Comment (reefdog) [inline README.md]:
```suggestion
3. Copy your Swagger specification:
```

Colon added for consistency w/ steps 1 and 2.

File: README.md [hunk 6]
```diff
@@ -2,30 +2,48 @@
...
+
+```shell
+cp YOUR_SWAGGER_SPEC_LOCATION ./openapi.json
+```
+
+4. Generate the code bases
```
Comment (reefdog) [inline README.md]:
```suggestion
4. Generate the codebases:
```

File: README.md (hunk 2 above)
Comment (slifty) [reply README.md]: I agree!  BUT DID NOT LAUGH.

File: README.md [hunk 7]
```diff
@@ -2,30 +2,48 @@
...
 ```shell
-> cp YOUR_SWAGGER_SPEC_LOCATION ./openapi.json
 > ./gradlew generateSwaggerCode
 ```
 
 The generated code will now exist in the `build` directory.
-Each language target will have its own directory inside of `build` (e.g. `build/typescript`).
-Some languages might have additional build steps defined in their build's README.md file.
+
+Each language target will have its own directory inside of `build`, and you can follow the readme instructions in that directory with additional steps related to that code base.
```
Comment (reefdog) [inline README.md]:
```suggestion
Each language target will have its own directory inside of `build/`, and you can follow the readme instructions in that directory with additional steps related to that codebase.
```

Including the trailing slash is a convention I've adopted (and like!) when discussing directories.

**Review by reefdog — CHANGES_REQUESTED** [review]
This is great and an improvement! I left a ton of little suggestions around things like case consistency. I think this will be a quick approval after that!

File: README.md (hunk 2 above)
Comment (reefdog) [reply README.md]: Dan how many tickles does it take to make an octopus laugh

File: README.md (hunk 6 above)
Comment (slifty) [reply README.md]: image

:-|

Comment (slifty) [reply README.md]: I decided to ditch "codebase" or "code base" entirely

Comment (reefdog) [reply README.md]: …weird. I'd say that's a dictionary with insufficient domain knowledge!

FWIW Wikipedia implies both are correct — "codebase (or code base)" — but goes on to use "codebase" throughout the article: https://en.wikipedia.org/wiki/Codebase

Comment (slifty) [reply README.md]: yeah the consensus is all of it is ok (including `code-base`) but I just realized the word is longer than needed anyway, I'm not even sure it's the correct term in this case.

(the project isn't swagger-codebasegen it's swagger-codegen!)

**Review by reefdog — APPROVED** [review]

**Pull request merged**